src/nurse/tests/test_views.py

```diff
@@ -784,3 +784,55 @@ def test_post_only_staff(self, user, client):
         assert response.json() == {
             "detail": "You do not have permission to perform this action."
         }
+
+
+@pytest.fixture
+def user_profile(user):
+    return UserOneSignalProfile.objects.create(user=user, subscription_id="123456789")
+
+
+@pytest.fixture
+def another_user_profile(another_user):
+    return UserOneSignalProfile.objects.create(
+        user=another_user, subscription_id="987654321"
+    )
+
+
+@pytest.fixture
+def another_user(db):
+    return User.objects.create_user(
+        username="username2", email="user2@example.com", password="testpass123"
+    )
+
+
+@pytest.mark.django_db
+class TestUserOneSignalProfileView:
+    url = reverse_lazy("useronesignalprofile-list")
+
+    def test_endpoint_onesignal(self):
+        assert self.url == "/onesignal/"
+
+    def test_create_onesignal(self, client, user):
+        data = {"subscription_id": "1233456789"}
+        client.force_authenticate(user=user)
+        response = client.post(self.url, data, format="json")
+        assert response.status_code == status.HTTP_201_CREATED
+        print(response.data)
+        assert UserOneSignalProfile.objects.count() == 1
+        assert response.data["user"] == user.id
+
+    def test_read_onesignal_single(self, client, user, user_profile):
+        client.force_authenticate(user=user)
+        response = client.get(self.url)
+        assert response.status_code == status.HTTP_200_OK
+        assert len(response.data) == 1
+        assert response.data[0]["subscription_id"] == user_profile.subscription_id
```

Comment:
I think the `test_read_onesignal_list` does more that this one, so we could remove this test and only keep `test_read_onesignal_list`

Reply:
Which do you want delete ? test_read_onesignal_list or test_read_onesignal_single

Reply:
Remove `test_read_onesignal_single` because it demonstrates less than `test_read_onesignal_list`

Reply:
I deleted `test_read_onesignal_single `